openmc/stats/univariate.py

```diff
@@ -1199,3 +1230,68 @@ def from_xml_element(cls, elem):
             distribution.append(Univariate.from_xml_element(pair.find("dist")))
 
         return cls(probability, distribution)
+
+    def integral(self):
+        """Return integral of the distribution
+
+        .. versionadded:: 0.13.1
+
+        Returns
+        -------
+        float
+            Integral of the distribution
+        """
+        return sum([
+            p*dist.integral()
+            for p, dist in zip(self.probability, self.distribution)
+        ])
+
+
+def combine_distributions(dists, probs):
+    """Combine distributions with specified probabilities
+
+    This function can be used to combine multiple instances of
+    :class:`~openmc.stats.Discrete` and `~openmc.stats.Tabular` into a single
+    distribution. Multiple discrete distributions are merged into a single
+    distribution and the remainder of the distributions are put into a
+    :class:`~openmc.stats.Mixture` distribution.
+
+    .. versionadded:: 0.13.1
+
+    Parameters
+    ----------
+    dists : iterable of openmc.stats.Univariate
+        Distributions to combine
+    probs : iterable of float
+        Probability (or intensity) of each distribution
+
+    """
+    # Get copy of distribution list so as not to modify the argument
+    dist_list = deepcopy(dists)
+
+    # Get list of discrete/continuous distribution indices
+    discrete_index = [i for i, d in enumerate(dist_list) if isinstance(d, Discrete)]
+    cont_index = [i for i, d in enumerate(dist_list) if isinstance(d, Tabular)]
+
+    # Apply probabilites to continuous distributions
+    for i in cont_index:
+        dist = dist_list[i]
+        dist.p *= probs[i]
+
+    if discrete_index:
+        # Create combined discrete distribution
+        dist_discrete = [dist_list[i] for i in discrete_index]
+        discrete_probs = [probs[i] for i in discrete_index]
+        combined_dist = Discrete.merge(dist_discrete, discrete_probs)
+
+        # Replace multiple discrete distributions with merged
+        for idx in reversed(discrete_index):
+            dist_list.pop(idx)
+        dist_list.append(combined_dist)
+
+    # Combine discrete and continuous if present
+    if len(dist_list) > 1:
+        probs = [d.integral() for d in dist_list]
```

Comment:
I think this line is the only one that stops someone from putting `Univariate` distributions other than `Discrete` and `Tabular` into this function. To me it makes sense for other distributions to have `integral()` methods from a purely mathematical standpoint so this behavior might be a bit unexpected for the user.

Reply:
That's a good point. From a practical standpoint, right now we only encounter tabular/discrete distributions in the decay data which was why I had only bothered to write `integral()` for those distributions. However, if you think it's work it, I could extend this to work with other distributions.

Reply:
I don't necessarily think we need to add `integral()` to every univariate distribution in the code as part of this PR, but I think it might be good eventually. Same with callables maybe for `pdf()` and `cdf()`

Reply:
Sounds good. Some cleanup with `pdf` and `cdf` is very much needed too although that is definitely outside the scope of this PR. If you're OK, let's save those for another time -- do you have any other requests for this PR?

Reply:
Nope!